javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,21 @@
+// query for all elements that are span
+// store the nodelist in a variable
+// when used in keyboard event:
+// event.key = returns the value of the key pressed
+//
+
+var $characters = document.querySelectorAll('span');
+var counter = 0;
+window.addEventListener('keydown', handleKeydown);
+
+function handleKeydown(event) {
+  if (($characters[counter].textContent === event.key) && (counter !== $characters.length - 1)) {
+    $characters[counter].className = 'correct';
+    $characters[counter + 1].className = 'border-bottom';
+    counter++;
+  } else if (($characters[counter].textContent !== event.key) && (counter !== $characters.length - 1)) {
+    $characters[counter].className = 'incorrect';
+  } else if (counter === $characters.length - 1) {
+    event.preventDefault();
+  }
```

Comment:
What happens if you delete this block?

Reply:
I get an error message if the user continues to type after the last space, so I thought I should "end" the event. And this was the only method I could think of!

Reply:
last space = last character*